modules/packages/OrderedSet/Treap.chpl

```diff
@@ -0,0 +1,862 @@
+                      default-initialized, here: " + eltType: string);
+      }
+      
+      _enter(); defer _leave();
+
+      var node = _lower_bound(_root, e);
+      var result: (bool, eltType);
+      if node == nil then  result[0] = false;
+      else result = (true, node!.element);
+      return result;
+    }
+
+    /*
+      Find the first element in the orderedSet
+      which does not compare less than e.
+
+      Returns a tuple containing two element:
+      The first element is a `bool` that indicates whether there is such an element.
+      The second element is the occurence in the orderedSet, if there's any.
+
+      :returns: a tuple containing result
+      :rtype: `(bool, eltType)`
+    */
+    proc const upperBound(e: eltType): (bool, eltType) {
+      if !isDefaultInitializable(e) {
+        compilerError("upperBound is not avaliable on types that can't be \
+                      default-initialized, here: " + eltType: string);
+      }
+
+      _enter(); defer _leave();
+
+      var node = _upper_bound(_root, e);
+      var result: (bool, eltType);
+      if node == nil then  result[0] = false;
+      else result = (true, node!.element);
+      return result;
+    }
+
+    /*
+      Find the predecessor of one element in the orderedSet.
+      Returns if there is such one element.
+
+      Returns a tuple containing two element:
+      The first element is a `bool` that indicates whether there is such an element.
+      The second element is the occurence in the orderedSet, if there's any.
+
+      :arg e: The element to base
+      :type e: `eltType`
+
+      :returns: a tuple containing result
+      :rtype: `(bool, eltType)`
+    */
+    proc const predecessor(e: eltType, ref result: eltType): (bool, eltType) {
+      if !isDefaultInitializable(e) {
+        compilerError("predecessor is not avaliable on types that can't be \
+                      default-initialized, here: " + eltType: string);
+      }
+
+      _enter(); defer _leave();
+
+      var result: (bool, eltType);
+      result[0] = false;
+      var baseNode = _find(_root, e);
+      if baseNode == nil {
+        return result;
+      }
+
+      var resultNode = _neighbour(baseNode, 0);
+      if resultNode != nil {
+        result = (true, resultNode!.element);
+      }
+      return result;
+    }
+    
+    /*
+      Find the successor of one element in the orderedSet.
+      Returns if there is such one element.
+      If there is, store the result in `result`.
+
+      Returns a tuple containing two element:
+      The first element is a `bool` that indicates whether there is such an element.
+      The second element is the occurence in the orderedSet, if there's any.
+
+      :arg e: The element to base
+      :type e: `eltType`
+
+      :returns: a tuple containing result
+      :rtype: `(bool, eltType)`
+    */
+    proc const successor(e: eltType, ref result: eltType): (bool, eltType) {
```

Comment:
Ref?

Reply:
Strange. I remember I've updated it. Maybe I mistook something about git

Reply:
https://github.com/chapel-lang/chapel/pull/16124/commits/26f4adca1da7cbc8b63f3e8a56f958b34288226e